relay-event-schema/src/protocol/span.rs

```diff
@@ -59,7 +59,7 @@ pub struct Span {
 
     /// Arbitrary tags on a span, like on the top-level event.
     #[metastructure(pii = "maybe")]
-    pub tags: Annotated<Object<JsonLenientString>>,
+    pub tags: Annotated<Tags>,
```

Comment:
The data structure changes from a map to an array. Is this change supported downstream (e.g. consumers)?

Reply:
Also -- should we skip serialization if there are no tags? `skip_serialization = "empty"`.

Reply:
This field is not set by current SDKs. It was added for a span-only world where we'd set tags directly on spans. Now, this is not propagated directly downstream as it's transformed to the `SpanKafkaMessage` first I modified in this PR. So, yes, the changes needed to support this type change are done.

@jjbayer do you know of any other use of that field outside of standalone spans?

We can indeed skip the serialization if it's empty.

Reply:
Looks great, thank you!